From f7066aa2de867e4ec58d9f8f4f3e4f2a434b3cde Mon Sep 17 00:00:00 2001
Message-Id: <f7066aa2de867e4ec58d9f8f4f3e4f2a434b3cde.1425610844.git.chang-joon.lee@intel.com>
In-Reply-To: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
References: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Fri, 13 Feb 2015 13:04:02 +0530
Subject: [PATCH 20/27] MUST_REBASE [VPG]: video/adf/intel: Fix DSI panel
 power on/off sequence

With VBT version 3.0, MIPI panel power on and MIPI panel power off has been
added as a sequence in VBT block#53. Correcting the logic of parsing of the
sequence.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-6003
Change-Id: I38fb11b658dfd529071e5ffc83ecdf06eeadb728
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 .../core/common/dsi/panels/intel_dsi_panel_vbt.c   |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c b/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
index 300f323..002b628 100644
--- a/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
+++ b/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
@@ -508,12 +508,15 @@ static u8 *mipi_exec_spi(struct dsi_pipe *dsi_pipe, u8 *data)
 
 static u8 *mipi_exec_pmic(struct dsi_pipe *dsi_pipe, u8 *data)
 {
-	u8 pmic_page;
 	u32 register_address, register_data;
 	u32 data_mask, tmp;
 
-	data++;
-	pmic_page = *data++;
+	/*
+	 * First 3 bytes are not relevant for Linux.
+	 * Skipping the data field by 3 bytes to get
+	 * the PMIC register Address.
+	 */
+	data += 3;
 	register_address = *((u32 *)data);
 	data += 4;
 	register_data = *((u32 *)data);
-- 
1.7.9.5

